champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/data/BullsEyeData.java

```diff
@@ -0,0 +1,107 @@
+    public LivingEntity getTarget() {
+        return target;
+    }
+
+    public ChargeData getTargetFocused() {
+        return targetFocused;
+    }
+
+    public void setTarget(LivingEntity newTarget) {
+        target = newTarget;
+    }
+
+    public void setTargetFocused(ChargeData newTargetFocused) {
+        targetFocused = newTargetFocused;
+    }
+
+    public void spawnFocusingParticles() {
+        float targetFocusedAmount = getTargetFocused().getCharge();
+
+        Location casterLocation = caster.getLocation().add(0, caster.getHeight() / 3, 0);
+        Location targetLocation = target.getLocation().add(0, target.getHeight() / 3, 0);
+
+        Vector direction = targetLocation.toVector().subtract(casterLocation.toVector()).normalize();
+        Vector rotatedDirection = new Vector(-direction.getZ(), direction.getY(), direction.getX()).normalize();
+
+        double circleRadius = getRadius(targetFocusedAmount);
+        double offsetX = circleRadius * rotatedDirection.getX();
+        double offsetY = circleRadius * rotatedDirection.getY();
+        double offsetZ = circleRadius * rotatedDirection.getZ();
+
+        Location particleLocation = targetLocation.clone().subtract(offsetX, offsetY, offsetZ).subtract(direction.multiply(2));
+
+        for (double angle = 0; angle < Math.PI * 2; angle += Math.PI / 10) {
+            Vector offset = rotatedDirection.clone().multiply(circleRadius * Math.cos(angle));
+            offset.setY(Math.sin(angle) * circleRadius);
+            particleLocation.add(offset);
+            caster.spawnParticle(Particle.REDSTONE, particleLocation, 1, new Particle.DustOptions(getColor(), 1));
+        }
+    }
+
+    private double getRadius(double targetFocusedAmount) {
+        return 0.5 - (targetFocusedAmount / 5);
+    }
+
+    public Color getColor() {
+        float targetFocusedAmount = targetFocused.getCharge();
+
+        if (targetFocusedAmount <= 0.09) {
+            return Color.fromRGB(250, 50, 50);
+        } else if (targetFocusedAmount <= 0.18) {
+            return Color.fromRGB(240, 65, 40);
+        } else if (targetFocusedAmount <= 0.27) {
+            return Color.fromRGB(240, 100, 40);
+        } else if (targetFocusedAmount <= 0.36) {
+            return Color.fromRGB(240, 130, 40);
+        } else if (targetFocusedAmount <= 0.45) {
+            return Color.fromRGB(240, 165, 40);
+        } else if (targetFocusedAmount <= 0.54) {
+            return Color.fromRGB(240, 200, 40);
+        } else if (targetFocusedAmount <= 0.63) {
+            return Color.fromRGB(235, 235, 40);
+        } else if (targetFocusedAmount <= 0.72) {
+            return Color.fromRGB(205, 240, 40);
+        } else if (targetFocusedAmount <= 0.81) {
+            return Color.fromRGB(170, 240, 40);
+        } else if (targetFocusedAmount <= 0.9) {
+            return Color.fromRGB(110, 240, 40);
+        } else if (targetFocusedAmount <= 1.0) {
+            return Color.fromRGB(70, 240, 40);
+        }
```

Comment:
Theres probably a cleaner way to do this

Perhaps multiply R (250) by 1-targetFocusedAmount, and multiply G (50) by 1+targetFocusedAmount

Reply:
Yeah, I tried a few ways but most seemed to go into dark and dull colours which didn't look too good, I'll try some other methods and see what I can come up with